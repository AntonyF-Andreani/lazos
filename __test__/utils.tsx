import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";

import theme from "theme";

/**
 * Colocar todos los providers, se puede mockear redux y las funciones
 */
const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement) => render(ui, { wrapper: AllTheProviders });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
