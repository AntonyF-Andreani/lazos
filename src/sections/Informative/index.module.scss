.root {
  text-align: center;
  margin-bottom: 50px;
}

.redSection,
.whiteSection {
  padding-block: 80px;

  @media only screen and (max-width: 768px) {
    :global(.swiper-wrapper) {
      width: auto;
    }

    padding: 30px;
  }
}

.redSection {
  background-color: #d71920;
}

.container {
  max-width: 1200px;
  margin: auto;
}

.flexContainer {
  display: flex;
}

.title {
  letter-spacing: -0.05em;
  font-weight: 700;
  font-size: 2.3em;
  color: #fff;
  margin-bottom: 60px;

  @media (max-width: 425px) {
    font-size: 2em;
  }
}

.description {
  width: 100%;
  columns: 2;
  font-size: 16px;
  font-weight: 300;
  line-height: 1.7;
  color: #e0e0e0;
  text-align: start;
  margin: auto;
  padding-inline: 15px;

  @media only screen and (max-width: 768px) {
    columns: 1;
    text-align: center;
  }

  p {
    margin: 0;
    padding: 0;

    @media only screen and (max-width: 768px) {
      margin-bottom: 20px;
    }
  }
}

.descWhite {
  font-size: 1.2em;
  color: #777;
  line-height: 26px;
  margin-inline: 40px;
  text-align: left;

  @media only screen and (max-width: 900px) {
    text-align: center;
    margin-inline: 0;
  }

  p {
    margin-bottom: 0;
  }
}

.titleWhite {
  color: #212529;
  letter-spacing: -0.05em;
  font-weight: 400;
  font-size: 2em;
  margin-inline: 40px;
  text-align: left;
  margin-top: 40px;

  @media (max-width: 375px) {
    margin-inline: 20px;
    font-size: 1.5em;
  }

  @media (max-width: 900px) {
    text-align: center;
  }

  @media (max-width: 1000px) {
    margin-top: 0;
  }

  p {
    margin-block: 0;
  }
}

.image {
  width: 1000px;
  padding-inline: 15px;

  @media only screen and (max-width: 900px) {
    display: none;
  }

  @media only screen and (max-width: 1000px) {
    width: 600px;
  }
}
