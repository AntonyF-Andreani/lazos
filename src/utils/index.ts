import { ImageLoaderProps } from "next/image";

/**
 * Utilidad para extraerla URL con formatos optimizados para las imagenes del CMS (Servicio de Imagenes)
 * https://www.storyblok.com/docs/image-service
 */
export const StoryBlockImageURL = ({
  src,
  width,
  quality = 20,
  format = "webp",
  height = 0,
}: StoryBlockImageURLParams) => {
  return `${src}/m/${width}x${height}/filters:quality(${quality}):format(${format})`;
};

interface StoryBlockImageURLParams {
  /** URL de la imagen como la provee Storyblok */
  src: string;
  /** Width of image. The Storyblok API infers aspect ratio */
  width: number;
  /** Quality of image, for default its 20 (20%). */
  quality?: number;
  /** Format of image, for default its webp */
  format?: "webp" | "jpeg" | "png";
  /** Height of the image */
  height?: number;
}

export const StoryBlockLoader = ({ src, width, quality = 20 }: ImageLoaderProps) => {
  return StoryBlockImageURL({ src, width, quality });
};

/**
 * Util for join classes undefined or string and filter undefined
 * @param args classNames
 *
 *  @example const className = joinClasses("a", "b", undefined, "c") // "a b c"
 */
export const joinClasses = (...args: (string | undefined)[]): string => {
  return args.filter((x) => x).join(" "); // undefined is falsey
};

/**
 * Util to test diferrent viewports, must be used with jest.
 *
 * This util make a mock implementation of window.matchMedia
 *
 * @example prepareViewport("(max-width: 768px)")
 */
export const prepareViewport = (mediaQuery: string) => {
  Object.defineProperty(window, "watchMedia", {
    writable: true,
    configurable: true,
    value: mediaQuery,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    return {
      matches: query === mediaQuery,
      media: "",
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};
