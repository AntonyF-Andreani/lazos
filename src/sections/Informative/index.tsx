import React from "react";
import { Grid, Typography, Box, Grow } from "@mui/material";
import Image from "next/image";

import styles from "./index.module.scss";

import Markdown from "components/MarkDown";
import { joinClasses } from "utils";

export interface InformativeProps {
  title: string;
  description: string;
  title_white: string;
  description_white: string;
}

const Informative = ({ title, description, title_white, description_white }: InformativeProps) => {
  return (
    <Grid className={styles.root} id="programa">
      <Grid className={styles.redSection}>
        <Box className={styles.container}>
          <Grow in={true} style={{ transformOrigin: "0 0 0" }} {...{ timeout: 4000 }}>
            <Typography className={styles.title} color="secondary" variant="h2">
              {title}
            </Typography>
          </Grow>
          <Grow in={true} style={{ transformOrigin: "0 0 0" }} {...{ timeout: 4000 }}>
            <Typography className={styles.description} color="secondary" component="div">
              <Markdown>{description}</Markdown>
            </Typography>
          </Grow>
        </Box>
      </Grid>
      <Grid className={styles.whiteSection}>
        <Box className={joinClasses(styles.container, styles.flexContainer)}>
          <Grid className={styles.image}>
            <Image
              alt="EnvÃ­os solidarios"
              height={175}
              layout="responsive"
              src={"https://a.storyblok.com/f/63950/1063x466/ed67e1913c/map.jpg"}
              width={400}
            />
          </Grid>
          <Grid>
            <Grid>
              <Typography className={styles.titleWhite} color="secondary" variant="h2">
                <Markdown>{title_white}</Markdown>
              </Typography>
            </Grid>
            <Grid>
              <Typography className={styles.descWhite} color="secondary" component="div">
                <Markdown>{description_white}</Markdown>
              </Typography>
            </Grid>
          </Grid>
        </Box>
      </Grid>
    </Grid>
  );
};

export default Informative;
