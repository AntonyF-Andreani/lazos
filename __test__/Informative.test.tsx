import React from "react";

import { render } from "./utils";

import Informative from "sections/Informative";
import type { InformativeProps } from "sections/Informative";

const defaultProps: InformativeProps = {
  title: "title test",
  description: "description test",
  title_white: "title white test",
  description_white: "description white test",
};

describe("<Informative />", () => {
  it("Should render", () => {
    const { asFragment } = render(<Informative {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("Should render title in red section", () => {
    const { getByText } = render(<Informative {...defaultProps} />);

    expect(getByText(defaultProps.title)).toBeInTheDocument();
  });
  it("Should render description in red section", () => {
    const { getByText } = render(<Informative {...defaultProps} />);

    expect(getByText(defaultProps.description)).toBeInTheDocument();
  });
  it("Should render title in white section", () => {
    const { getByText } = render(<Informative {...defaultProps} />);

    expect(getByText(defaultProps.title_white)).toBeInTheDocument();
  });
  it("Should render description in white section", () => {
    const { getByText } = render(<Informative {...defaultProps} />);

    expect(getByText(defaultProps.description_white)).toBeInTheDocument();
  });
});
