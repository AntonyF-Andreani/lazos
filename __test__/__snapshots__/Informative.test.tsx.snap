// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Informative /> Should render 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root root css-vj1n65-MuiGrid-root"
    id="programa"
  >
    <div
      class="MuiGrid-root redSection css-vj1n65-MuiGrid-root"
    >
      <div
        class="container MuiBox-root css-0"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 title css-rpv6ho-MuiTypography-root"
          style="opacity: 1; transform: scale(1, 1); transform-origin: 0 0 0; transition: opacity 4000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 2664ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          title test
        </h2>
        <div
          class="MuiTypography-root MuiTypography-body1 description css-1uk7n8n-MuiTypography-root"
          style="opacity: 1; transform: scale(1, 1); transform-origin: 0 0 0; transition: opacity 4000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 2664ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <p>
            description test
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root whiteSection css-vj1n65-MuiGrid-root"
    >
      <div
        class="container flexContainer MuiBox-root css-0"
      >
        <div
          class="MuiGrid-root image css-vj1n65-MuiGrid-root"
        >
          <div
            style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="display: block; box-sizing: border-box; padding-top: 43.75%;"
            />
            <img
              alt="EnvÃ­os solidarios"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <div
            class="MuiGrid-root css-vj1n65-MuiGrid-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 titleWhite css-rpv6ho-MuiTypography-root"
            >
              <p>
                title white test
              </p>
            </h2>
          </div>
          <div
            class="MuiGrid-root css-vj1n65-MuiGrid-root"
          >
            <div
              class="MuiTypography-root MuiTypography-body1 descWhite css-1uk7n8n-MuiTypography-root"
            >
              <p>
                description white test
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
